### ISSUE TITLE
Bug: Delete at start of heading

### ISSUE THREAD

**Issue by affvalente** [facebook/lexical#4359]
When pressing delete at the start of heading, Lexical deletes the prior line as expected but resets the note to paragraph

```
Paragraph

## Heading
Paragraph
```

In this text, if the cursor is placed just left of the "H" and I hit delete, this becomes

```
Paragraph
Paragraph
Paragraph
```

Was able to replicate on the playground on windows+chrome

Desired behaviour: delete the prior empty line to achieve 

```
Paragraph
## Heading
Paragraph
```

**Comment by affvalente** [comment]
Any news on this one?

**Comment by levensta** [comment]
This is still relevant. Pressing Backspace deletes the previous character and reformats the HeadingNode (or any other ElementNode) into a paragraph @etrepum fyi

https://github.com/user-attachments/assets/d063b6bd-b3bf-4290-8aa6-21eaa97188ee

It seems that the Heading should first be reformatted into a Paragraph, and only when Backspace is pressed again should the `<br>` be deleted

https://playground.lexical.dev/#doc=H4sIAAAAAAAAE81TQWrDMBD8y5xFsINBVA8oPQZ8LD0s1sYWKJKR16Em-O9FTlrHvbUE2pM0s-zM7CJdwNZJTLWQMMwFKUbJZ9M5bxMHmNdvwLKQ8zCFwjGmE8lyPUXLMAiZ8VAYZPKZgILwu8DgQInaRH2XqanPxaWicOY0uBhgyvlNwbrEjSw4jN6vLlnLBcvh6njT6O9kV6Gr6_OaL8P6M9OsNjP9ienvF_nCZF1oH7vG7kt0Mw-1uVY-Lvx_egU_t1l-xybqrOBpkJrObGFKrXVVPhV7vdeVwhDH1OS-g6epTXEM9r4bxa6qdgXmD8pTnbWGAwAA

**Comment by etrepum** [comment]
What happens here in lexical is that removeText merges the ParagraphNode with the HeadingNode, which is the right behavior in most cases but many users expect a special case of removing the ParagraphNode instead when the ParagraphNode is empty (at least that's the convention in Google Docs).

**Comment by levensta** [comment]
Merge also occurs if there is a non-empty paragraph before the heading, but it ends with a line break

https://github.com/user-attachments/assets/75b7e1b0-c643-4602-9ecd-c8c476b1dbb8

**Comment by etrepum** [comment]
Yes, right now merge occurs unconditionally when there’s a previous block ElementNode.
